index.js

```diff
@@ -25,10 +25,11 @@ var KNOWN_STATICS = {
 };
 
 module.exports = function hoistNonReactStatics(targetComponent, sourceComponent) {
-    var keys = Object.getOwnPropertyNames(sourceComponent);
-    for (var i=0; i<keys.length; ++i) {
-        if (!REACT_STATICS[keys[i]] && !KNOWN_STATICS[keys[i]]) {
-            targetComponent[keys[i]] = sourceComponent[keys[i]];
+    var sourceKeys = Object.getOwnPropertyNames(sourceComponent),
+        targetKeys = Object.getOwnPropertyNames(targetComponent);
+    for (var i=0; i<sourceKeys.length; ++i) {
+        if (!REACT_STATICS[sourceKeys[i]] && !KNOWN_STATICS[sourceKeys[i]] && targetKeys.indexOf(sourceKeys[i]) === -1) {
```

Comment:
Interestingly, the very same code base (https://github.com/gaearon/react-proxy/blob/v2.0.1/src/createClassProxy.js#L33-L35) uses `Object.prototype.hasOwnProperty.call(InitialComponent, '__reactPatchProxy')` to check if property exists. Is there a reason `Object.prototype.hasOwnProperty.call` would behave different from `targetComponent.hasOwnProperty('__reactPatchProxy')`?